lib/ansible/modules/web_infrastructure/ansible_tower/tower_project.py

```diff
@@ -155,12 +186,18 @@ def main():
                     except (exc.NotFound) as excinfo:
                         module.fail_json(msg='Failed to update project, credential not found: {0}'.format(scm_credential), changed=False)
 
+                if (scm_update_cache_timeout is not None) and (scm_update_on_launch is not True):
+                    module.warn('scm_update_cache_timeout will be ignored since scm_update_on_launch was not set to true')
```

Comment:
In cases of an invalid combination of parameters, I would like to rely on server-side validation raising an error. However, it seems that this scenario might actually be accepted, so I would agree that either warning or error on the module side is nice, because otherwise user would be left without any indication.